core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -3034,7 +3034,8 @@ private void setBuildsAndWorkspacesDir() throws IOException, InvalidBuildsDir {
 
         String newWorkspacesDir = SystemProperties.getString(WORKSPACES_DIR_PROP);
         if (newWorkspacesDir != null && !workspaceDir.equals(newWorkspacesDir)) {
-            LOGGER.log(Level.WARNING, "Changing workspaces directories from {0} to {1}. Beware that no automated data migration will occur.",
+        	Level level = workspaceDir.equals(DEFAULT_WORKSPACES_DIR) ? Level.INFO : Level.WARNING;
```

Comment:
when will this be set to `INFO` and when will it be set to `WARNING`, what is the test condition in the surrounding block?